browser/gui/app.cpp

```diff
@@ -156,6 +157,35 @@ void App::set_scale(unsigned scale) {
     engine_.set_layout_width(windowSize.x / scale_);
 }
 
+void App::sticky_windows(sf::Event::SizeEvent size) {
+    const char *names[] = {
+        "Navigation",
+        "HTTP Response",
+        "DOM",
+        "Stylesheet",
+        "Layout",
+        NULL,
+    };
+    
+    int delta = k_width - size.width;
+    for (uint8_t i = 0;; ++i) {
+        const char *name = names[i];
+        if (!name) {
+            break;
+        }
+        ImGuiWindow *win = ImGui::FindWindowByName(name);
+        if (!win) {
```

Comment:
Isn't this a programming error if it ever happens? Until we have a saner way of keeping the names in sync, just `assert()` that we have the window we're looking for.

Reply:
> Isn't this a programming error if it ever happens? 

No, because we might decide that windows are ephemeral. I can think of plenty of reasons we might want to destroy some # of them at runtime.

Assert will crash the app, window stickiness should be a "best effort" feature. No reason to make the app user hostile when it doesn't need to be.

Reply:
Right now, all windows always exist. If that changes in the future it will be because a developer changes it and they probably need to have a look at this code and think about what they're doing.

Since you made me look at the imgui internals now, did you have a look at using `GImGui.Windows` for this so we don't have to hard-code a list of window names?

Reply:
yes/no?